ui/src/components/admin/faqs/FaqGrid.js

```diff
@@ -34,24 +86,217 @@ export default function FaqGrid(props) {
+      colApi.setColumnVisible('isDeleted', !isReorderingFaqs);
+    }
   };
 
+  // Set the filters when reordering FAQs
+  const setReorderFaqFilters = () => {
+    setShowPublishedOnly(isReorderingFaqs);
+    setPageTopicFilter(isReorderingFaqs ? pageTopicList[0] : '');
+  };
+
+  // FAQ Publishing & Reordering Functions
+  // Handles FAQ publishing/unpublishing
+  const publishHandler = async () => {
+    if (api) {
+      const selectedFaqs = api.getSelectedNodes();
+
+      if (selectedFaqs && selectedFaqs.length) {
+        let tempFaqs = deepCopyArray(localFaqs);
+
+        if (toPublish) {
+          selectedFaqs.forEach((faq) => {
+            const idx = tempFaqs.findIndex((f) => f.id === faq.data.id);
+            if (idx !== -1) tempFaqs[idx].isPublished = true;
+          });
+        }
+
+        if (toUnpublish) {
+          selectedFaqs.forEach((faq) => {
+            const idx = tempFaqs.findIndex((f) => f.id === faq.data.id);
+            if (idx !== -1) tempFaqs[idx].isPublished = false;
+          });
+        }
+
+        if (toPublish || toUnpublish) {
+          resetHandler();
+          setLocalFaqs(tempFaqs);
+          setIsReorderingFaqs(true);
+        }
+      } else {
+        toast({
+          description: 'No FAQs selected',
+          status: 'error',
+          duration: 3000,
+        });
+      }
+    }
+  };
+  // Enables FAQ reordering
+  const enableFaqReordering = () => {
+    setShowDeleted(false);
+    setIsReorderingFaqs(true);
+  };
+  // Cancels FAQ publishing/reordering
+  const cancelHandler = () => {
+    setIsReorderingFaqs(false);
+    // Revert local FAQs to database FAQs
+    setLocalFaqs(deepCopyArray(faqs));
+  };
+  // Set the order of the FAQs when a row is dragged
+  const onRowDragEnd = (params) => {
+    if (params.api) {
+      let tempFaqs = deepCopyArray(localFaqs);
+      params.api.forEachNode((node) => {
+        const idx = tempFaqs.findIndex((f) => f.id === node.data.id);
+        if (idx !== -1) tempFaqs[idx].order = node.rowIndex + 1;
+        
+      });
+      setLocalFaqs(tempFaqs);
+    }
+  };
+
+  // FAQ Update Functions
+  const saveFaqInfo = async (data) => {
+    return updateHandler(data);
+  };
+
+  const saveFaqReordering = async () => {
+    try {
+      const updatePromises = localFaqs.map(async (faq) => {
+        const toUpdate = {
+          id: faq.id,
+          order: faq.order,
+          isPublished: faq.isPublished,
+        };
+
+        const success = await saveFaqInfo(toUpdate);
+        if (!success) {
+          console.log('Error saving FAQ');
+        }
+      });
+      
+      // Wait for all FAQs to be updated
+      await Promise.all(updatePromises);
+
+      toast({
+        description: 'Successfully reordered FAQs',
+        status: 'success',
+        duration: 5000,
+      });
+
+      // Refresh FAQs with updated data
+      await getData();
+
+      setIsReorderingFaqs(false);
```

Comment:
copy these setters which i think is for initializing down to the catch case too. otherwise the frontend might be stuck when there's an error

Reply:
ok!